scripts/prepare.js

```diff
@@ -2,8 +2,18 @@ const fs = require("fs");
 const jsyaml = require("js-yaml");
 const mustache = require("mustache");
 const graph = require("@graphprotocol/graph-cli/src/cli");
+const commander = require("commander");
 
-const network = process.argv.slice(2)[0];
+commander
+  .option("-n, --network <network>", "Network")
+  .option("-s, --startBlock <startBlock>", "Start Block")
+  .option("-f, --filename <filename>", "Filename")
+  .parse(process.argv);
+const options = commander.opts();
+
+const network = options.network || undefined;
+const startBlock = parseInt(options.startBlock) || undefined;
+const filename = options.filename || "subgraph";
```

Comment:
i like the idea of using commander here, though i'm not sure it's worth the added complexity. questions on the added properties:

**startBlock** i'm still not sure if i see why contract startblocks should ever be set to the same value. if that block number is too late, important events will get missed like creation/activity of earlier projects. if it's too early then some contracts will be unnecessarily indexed before they've been deployed. i could imagine setting this value to a recent block number if you don't care about having complete subgraph data, but this might also lead to unexpected behavior or an invalid testing environment. in any case, the strategy for that should be to use grafting.

**filename** When would a file need to be named differently than `subgraph.yaml`?

Reply:
agreed we can throw a lot of this away esp. if you are already doing QoL overhauls. i ended up not using a lot of it.

Reply:
cool. let's leave this file unchanged for now for simplicity